openedx/core/djangoapps/user_api/legacy_urls.py

```diff
@@ -30,7 +30,7 @@
     url(
         r'^user_api/v1/preferences/email_opt_in/$',
         user_api_views.UpdateEmailOptInPreference.as_view(),
-        name="preferences_email_opt_in"
+        name="preferences_email_opt_in_legacy"
     ),
     url(
         r'^user_api/v1/preferences/time_zones/$',
```

Comment:
I think that all these urls should be actually moved from legacy_urls.py -- in other words: these urls should be copied to `openedx/core/djangoapps/user_api/urls.py` and then we should remove this legacy_urls module. I initially suggested [in my comment](openedx/core/djangoapps/user_api/urls.py) that the legacy urls be kept in a separate section, but you can also decide to groupe the new and old urls for better readability:

```
# comment 1
new url1
old url 1

# comment 2
new url2
old url 2
...
```

Reply:
Once this PR has been merged we can remove the URLs (which starts with `user_api/v1/`) from the `openedx/core/djangoapps/user_api/legacy_urls.py` but we can not move this (https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/user_api/legacy_urls.py#L19) to `openedx/core/djangoapps/user_api/urls.py` as both of these `user_api/legacy_urls` and `user_api/urls` are registered with different regex in `lms/urls.py` (https://github.com/edx/edx-platform/blob/master/lms/urls.py#L134L138). And with these 2 different regex there is no need for `new url` and `old url`. Let me know your thoughts on this.

Reply:
Right! I understand. The `legacy_urls.py` module is still required because it is included from the main `lms/urls.py` file. We cannot move the content of `openedx/core/djangoapps.user_api/legacy_urls.py` to `openedx/core/djangoapps/user_api/urls.py` because these two modules are included differently.